app/adapters/application.js

```diff
@@ -2,16 +2,33 @@ import Ember from 'ember';
 import DS from 'ember-data';
 import ENV from '../config/environment';
 
+// matches query params or trailing slash
+const TRAILING_SLASH_PATTERN = /((^.*?[^\/])\?(.*?)$|^[^\?]+?[^\/]$)/;
+
 export default DS.RESTAdapter.extend(DS.BuildURLMixin, {
+    session: Ember.inject.service(),
     namespace: 'api/v2',
     host: ENV.apiBaseUrl,
     headers: {
         Accept: 'application/json'
     },
     ajax(url, method, hash) {
+        // trailing slash required for POST requests
+        let urlRegex = new RegExp(TRAILING_SLASH_PATTERN);
+        if (url.match(urlRegex)) {
```

Comment:
Nope! The regex is not global, it scans left to right and stops after the first match.